### DISCUSSION TITLE
Github Actions: Combine prove and CodePolicy?

### DISCUSSION THREAD

**Discussion by svenoe** [RotherOSS/otobo#1110] [Ideas]
For CodePolicy I was not aware, that basically all cpan dependencies of the tested files will have to be installed (see https://github.com/RotherOSS/otobo/runs/2248680414?check_suite_focus=true). For resource efficiency I propose to do both tasks in one github action. We can just install CodePolicy in the docker-otobo of the prove action and run it there.

For the same reason I propose to run this resulting action only on pull requests and commits to the release branches. Any objections @bschmalhofer?

**Comment by bschmalhofer** [comment]
I'm not too happy with using the otobo-docker image for compile checking and for the CodePolicy, even though this was the easiest way when I wrote the prove action. A better way might be to create create an artefact with all the CPAN dependencies and copy that into the standard image. Alternatively we could create a slim image for testing that is the standard image plus the CPAN dependencies.

**Comment by bschmalhofer** [comment]
In rel-10_1 I have added a cache for the installed CPAN modules. This means that the workflows syntax_check.yml and code_policy.yml execute faster now. Combining the two workflows into a single workflow should improve performance as well. I propose to go for it.

**Comment by bschmalhofer** [comment]
The workflows are now merged for the branches rel-10_0 and rel-10_1.  See #1255.
